test/framework/src/main/java/org/elasticsearch/test/transport/MockTransportService.java

```diff
@@ -81,7 +81,7 @@
  * (for example, @see org.elasticsearch.discovery.HandshakingTransportAddressConnector, which constructs
  * fake DiscoveryNode instances where the publish address is one of the bound addresses).
  */
-public final class MockTransportService extends TransportService {
+public class MockTransportService extends TransportService {
```

Comment:
Drop final so Mockito can spy on it. There are ways to test without spy. But they are not as straightforward and more involved. I think the effort is not worth the benefit. Given this is a test only class, I think dropping the final is acceptable.